tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -0,0 +1,233 @@
+from wazuh_testing.end_to_end.configuration import backup_configurations, restore_backup, configure_environment
+from wazuh_testing.end_to_end.logs import truncate_agents_logs, truncate_managers_logs, truncate_logs
+from wazuh_testing.end_to_end.wazuh_api import  get_agents_vulnerabilities
+from wazuh_testing.end_to_end.waiters import wait_until_vd_is_updated, wait_until_vuln_scan_agents_finished
+from wazuh_testing.end_to_end.monitoring import generate_monitoring_logs_all_agent, monitoring_events_multihost
+from wazuh_testing.end_to_end.regex import get_event_regex
+from wazuh_testing.end_to_end.indexer_api import get_indexer_values
+from wazuh_testing.tools.system import HostManager
+
+
+TIMEOUT_SYSCOLLECTOR_SCAN = 200
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+local_path = os.path.dirname(os.path.abspath(__file__))
+current_dir = os.path.dirname(__file__)
+configurations_dir = os.path.join(current_dir, "data", "configurations")
+configurations_paths = {
+        'manager': os.path.join(configurations_dir, 'manager.yaml'),
+        'agent': os.path.join(configurations_dir, 'agent.yaml')
+        }
+
+
+def load_vulnerability_detector_configurations():
+    """Return the configurations for Vulnerability testing for the agent and manager roles
+
+    Return:
+        Dict: Configurations for each role
+    """
+    return {
+        'agent': load_configuration_template(configurations_paths['agent'], [{}], [{}]),
+        'manager': load_configuration_template(configurations_paths['manager'], [{}], [{}])
+    }
+
+
+@pytest.fixture(scope='module')
+def setup_vulnerability_tests(host_manager: HostManager) -> Generator:
+    """Setup the vulnerability tests environment
+
+    Args: 
+        host_manager (HostManager): An instance of the HostManager class.
+    """
+    logger.error("Init setup of environment")
+
+    # Configure managers and agents
+    logger.error("Getting backup of current configurations")
+    hosts_configuration_backup = backup_configurations(host_manager)
+    logger.error("Configuring environment")
+    configure_environment(host_manager, load_vulnerability_detector_configurations())
+
+    # Restart managers and stop agents
+    logger.error("Stopping agents")
+    host_manager.control_environment('stop', ['agent'])
+    logger.error("Restarting managers")
+    host_manager.control_environment('restart', ['manager'])
+
+    # Wait until VD is updated
+    logger.error("Wait until Vulnerability Detector has update all the feeds")
+    wait_until_vd_is_updated(host_manager)
+
+    # Truncate alerts and logs of managers and agents
+    logger.error("Truncate managers and agents logs")
+    truncate_logs(host_manager)
+
+    # Start agents
+    logger.error("Starting agents")
+    host_manager.control_environment('start', ['agent'])
+
+    yield
+
+    logger.error("Restoring original configuration")
+    restore_backup(host_manager, hosts_configuration_backup)
```

Comment:
Should we truncate logs and alerts again so the environment is clean?

Reply:
Done feb16469d0897ad3404c7d744d43bc71d244a40d